samples/Web/deploy-webapp-ip-restrictions/azurepolicy.json

```diff
@@ -0,0 +1,75 @@
+{
+    "properties": {
+        "displayName": "Web App default IP restrictions",
+        "mode": "indexed",
+        "description": "Configures Web App IP restriction to allow access from specified public IP addresses",
+        "metadata": {
+            "category": "Web"
+        },
+        "parameters": {},
+        "policyRule": {
+            "if": {
+                "field": "type",
+                "equals": "Microsoft.Web/sites"
+            },
+            "then": {
+                "effect": "DeployIfNotExists",
+                "details": {
+                    "type": "Microsoft.Web/sites/config",
+                    "roleDefinitionIds": [
+                        "/providers/Microsoft.Authorization/roleDefinitions/de139f84-1756-47ae-9be6-808fbbe84772"
+                    ],
+                    "existenceCondition": {
+                        "not": {
+                            "field": "Microsoft.Web/sites/config/web.ipSecurityRestrictions[*].action",
+                            "Equals": "Allow"
+                        }
+                    },
+                    "deployment": {
+                        "properties": {
+                            "mode": "incremental",
+                            "parameters": {
+                                "name": {
+                                    "value": "[field('name')]"
+                                }
+                            },
+                            "template": {
+                                "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                                "contentVersion": "1.0.0.0",
+                                "parameters": {
+                                    "name": {
+                                        "type": "string"
+                                    }
+                                },
+                                "resources": [
+                                    {
+                                        "name": "[concat(parameters('name'), '/web')]",
+                                        "type": "Microsoft.Web/sites/config",
+                                        "apiVersion": "2018-11-01",
+                                        "properties": {
+                                            "scmIpSecurityRestrictionsUseMain": true,
+                                            "ipSecurityRestrictions": [
```

Comment:
>"ipSecurityRestrictions": [ [](start = 44, length = 27)

As written this deployment would blow away any existing ipSecurityRestrictions in the config.  If it isn't possible to merge them into the existing array I don't think this is a good candidate for a sample policy.

Reply:
Yes, I agree.  This policy is intended to apply a default set of ipRestrictions if there are none.

Reply:
Would it be possible to rework this sample policy so that it would audit if the user has specified ipSecurityRestrictions and maybe even deny the deploy if nothing is specified? I am currently trying to do that based on what is outlined in this sample and wanted your thoughts on changing up the sample.